CondFormats/HGCalObjects/interface/HGCalMappingCellIndexerTrigger.h

```diff
@@ -0,0 +1,179 @@
+                                         << ") is larger than allocated";
+    return di_[idx];
+  }
+
+  /**
+    @short builders for the dense index
+  */
+  uint16_t denseIndex(std::string typecode, uint32_t ROC, uint32_t trLink, uint32_t trCell) const {
+    return denseIndex(getEnumFromTypecode(typecode), ROC, trLink, trCell);
+  }
+  uint32_t denseIndex(size_t idx, uint32_t ROC, uint32_t trLink, uint32_t trCell) const {
+    return di_[idx].denseIndex({{ROC, trLink, trCell}}) + offsets_[idx];
+  }
+
+  /**
+     @short decodes the dense index code
+  */
+  /* ###############################
+   Only for Data, Trigger does not have elecId yet
+  uint32_t elecIdFromIndex(uint32_t rtn, std::string typecode) const {
+    return elecIdFromIndex(rtn, getEnumFromTypecode(typecode));
+  }
+  uint32_t elecIdFromIndex(uint32_t rtn, size_t idx) const {
+    if (idx >= di_.size())
+      throw cms::Exception("ValueError") << " index requested for cell dense indexer (i=" << idx
+                                         << ") is larger than allocated";
+    rtn -= offsets_[idx];
+    auto rtn_codes = di_[idx].unpackDenseIndex(rtn); // {}
+    return HGCalElectronicsId(false, 0, 0, 0, rtn_codes[0], rtn_codes[1]).raw();
+  }
+  ################################*/
+
+  /**
+     @short returns the max. dense index expected
+   */
+  uint32_t maxDenseIndex() const {
+    size_t i = maxTrLink_.size();
+    if (i == 0)
+      return 0;
+    return offsets_.back() + maxROC_.back() * maxTrLink_.back() * maxTCPerLink_.back();
+  }
+
+  /**
+     @short gets the number of words (cells) for a given typecode
+     Note : some partials are rounded to the closest multiplie of 16 or 8 depending on the density
+     That is done because not all the TrigLinks,TrgCells possible are assigned in practice
+     e.g.: ML-T has 22 TCs but this will return 32 or MH-T has 19 but it will return 24
+     It results in a small mem overhead over the totall memory needed to be allocated
+  */
+  size_t getNWordsExpectedFor(std::string typecode) const {
+    auto it = getEnumFromTypecode(typecode);
+    return getNWordsExpectedFor(it);
+  }
+  size_t getNWordsExpectedFor(size_t typecodeidx) const { return getDenseIndexerFor(typecodeidx).maxIndex(); }
+
+  /**
+     @short gets the number of Trigger Links for a given typecode
+  */
+  size_t getNTrLinkExpectedFor(std::string typecode) const {
+    auto it = getEnumFromTypecode(typecode);
+    return getNTrLinkExpectedFor(it);
+  }
+  size_t getNTrLinkExpectedFor(size_t typecodeidx) const { return maxTrLink_[typecodeidx] * maxROC_[typecodeidx]; }
+
+  constexpr static char maxHalfPerROC_ = 2;
+
+  std::map<std::string, size_t> typeCodeIndexer_;
+  std::vector<uint16_t> maxROC_;
+  std::vector<uint16_t> maxTrLink_;
+  std::vector<uint16_t> maxTCPerLink_;
+  std::vector<uint32_t> offsets_;
+  std::vector<HGCalDenseIndexerBase> di_;
+
+  ~HGCalMappingCellIndexerTrigger() {}
```

Comment:
```suggestion
  ~HGCalMappingCellIndexerTrigger() = default;
```

Reply:
thanks I'll address them and take the chance to squash the PR

Reply:
OK